src/alley/wp/simplecache/class-metadata-adapter.php

```diff
@@ -0,0 +1,222 @@
+	) {}
+
+	/**
+	 * Create an instance using the default composition.
+	 *
+	 * @param string $type Object type.
+	 * @param int    $id   Object ID.
+	 * @return CacheInterface
+	 */
+	public static function create( string $type, int $id ): CacheInterface {
+		return new PSR16_Compliant(
+			new NativeClock(),
+			new Prefixed_Keys(
+				'_psr16_',
+				new Maximum_Key_Length(
+					255,
+					new self(
+						$type,
+						$id,
+					),
+				),
+			),
+		);
+	}
+
+	/**
+	 * Create an instance for caching to post meta.
+	 *
+	 * @param int $post_id Post ID.
+	 * @return CacheInterface
+	 */
+	public static function for_post( int $post_id ): CacheInterface {
+		return self::create( 'post', $post_id );
+	}
+
+	/**
+	 * Create an instance for caching to term meta.
+	 *
+	 * @param int $term_id Term ID.
+	 * @return CacheInterface
+	 */
+	public static function for_term( int $term_id ): CacheInterface {
+		return self::create( 'term', $term_id );
+	}
+
+	/**
+	 * Fetches a value from the cache.
+	 *
+	 * @throws \Psr\SimpleCache\InvalidArgumentException If the $key string is not a legal value.
+	 *
+	 * @param string $key     The unique key of this item in the cache.
+	 * @param mixed  $default Default value to return if the key does not exist.
+	 * @return mixed The value of the item from the cache, or $default in case of cache miss.
+	 */
+	public function get( string $key, mixed $default = null ): mixed {
+		$get = $this->getMultiple( [ $key ], $default );
+
+		$out = $default;
+
+		foreach ( $get as $value ) {
+			$out = $value;
+		}
+
+		return $out;
+	}
+
+	/**
+	 * Persists data in the cache, uniquely referenced by a key with an optional expiration TTL time.
+	 *
+	 * @throws \Psr\SimpleCache\InvalidArgumentException If the $key string is not a legal value.
+	 *
+	 * @param string                 $key   The key of the item to store.
+	 * @param mixed                  $value The value of the item to store, must be serializable.
+	 * @param null|int|\DateInterval $ttl   Optional. The TTL value of this item.
+	 * @return bool True on success and false on failure.
+	 */
+	public function set( string $key, mixed $value, \DateInterval|int|null $ttl = null ): bool {
+		return $this->setMultiple( [ $key => $value ], $ttl );
+	}
+
+	/**
+	 * Delete an item from the cache by its unique key.
+	 *
+	 * @throws \Psr\SimpleCache\InvalidArgumentException If the $key string is not a legal value.
+	 *
+	 * @param string $key The unique cache key of the item to delete.
+	 * @return bool True if the item was successfully removed. False if there was an error.
+	 */
+	public function delete( string $key ): bool {
+		if ( ! $this->has( $key ) ) {
+			return true;
+		}
+
+		return (bool) delete_metadata( $this->type, $this->id, $key );
+	}
+
+	/**
+	 * Wipes clean the entire cache's keys.
+	 *
+	 * @return bool True on success and false on failure.
+	 */
+	public function clear(): bool {
+		return false; // Not supported by default in WordPress.
```

Comment:
I think this isn't supported by default _in memcached_ but there is a way to do this in Redis. Fine to keep as-is for now but we may want to consider a future enhancement where we could keep an inventory of cache keys that could be deleted on a clear operation (maybe in a separate database table even). That's a common enough problem that we run into where a generic solution would be excellent.

Reply:
As discussed in Slack, this is "not supported" in the sense that there is no built-in way to delete only the meta keys that have been saved by this library. The library could add support for `clear()` by adding its own manifest of cache keys that would then be deleted.